### ISSUE TITLE
Making `k6/experimental/browser` module part of k6 core

### ISSUE THREAD

**Issue by ankur22** [grafana/xk6-browser#1117]
## What

Migrate the browser module out of experimental:

`k6/experimental/browser` -> `k6/browser`

## Why

For this to be achievable the aim is to make the browser module:

1. stable during the runtime;
2. have a stable API.

Some of our users will not want to work with an experimental module with the fear that the cost of maintaining many scripts and breaking changes out weighs the pros of working with the browser module. Graduating the browser module out of experimental will signal to those users that:

1. The APIs are backwards compatible, and the upmost effort will be put into ensuring we don't break this;
2. No major breaking changes are coming to the APIs;
3. There are no known major issues.

## Blockers

We need to finalize and possibly work on the following points before we can be ready to graduate browser out of experimental:

```[tasklist]
### Tasks
- [ ] https://github.com/grafana/xk6-browser/issues/428
- [ ] https://github.com/grafana/xk6-browser/issues/1138
- [ ] https://github.com/grafana/xk6-browser/issues/1292
- [ ] Update the roadmap.
- [ ] Is xk6-browser going to support multiple browsers other than chromium/chrome?
- [ ] Are there any major refactors coming up that could destabilize the module?
- [ ] How will k6 v1 affect browser module's stability?
- [x] Should browser options be shortened ([#3310](https://github.com/grafana/k6/issues/3310))? **Not now, we need more feedback.**
- [x] Do all xk6-browser APIs need to be converted to async? **Yes they do**
- [x] Does xk6-browser need to be more compatible with Playwright APIs?
- [x] If we haven't reached enough of a compatibility with Playwright, what is enough?
- [x] Are we expecting any major breaking changes?
- [x] Are there any major issues that must be resolved (e.g. [#971](https://github.com/grafana/xk6-browser/issues/971))?
```

**Comment by inancgumus** [comment]
Nice list of tasks 👍 I'd also like to add the following.

> Are there any major refactors coming up that could destabilize the module?

I wouldn't say _"to destabilize"_. But:
* We have [a few big refactorings in the pipeline](https://github.com/grafana/xk6-browser/issues?q=is%3Aopen+is%3Aissue+label%3Arefactor).
* We also might have #228 wrapped up for stabilizing and cementing our code choices. This can also help us fearlessly refactor as well.
* We also have this epic issue: https://github.com/grafana/xk6-browser/issues/688.

> Do all xk6-browser APIs need to be converted to async? [#428](https://github.com/grafana/xk6-browser/issues/428)

There are open issues other than #428 tagged with `#async` that we might want to discuss: https://github.com/grafana/xk6-browser/labels/async.

> If we haven't reached enough of a compatibility with Playwright, what is enough?

This issue tracks PW compatibility #784. Marie has an updated document based on this issue that we can work on to decide what is enough.

**Comment by inancgumus** [comment]
Closed by grafana/k6#3793
